### PR TITLE
Add netcore50 source build for System.Reflection.Context

### PR DESCRIPTION
**Pull request by mellinoe**
We don't have any tests for this library, primarily because this is just a stub, all of the members throw PlatformNotSupportedException. I did some minor cleanup of the single source file in the second commit.

There is a .NET Framework version of this library, but it shipped in-box, so we don't need to build anything here for the NuGet package (other than the contract assembly which is already here).

@weshaggard

### CODE REVIEW

File: src/System.Reflection.Context/src/System.Reflection.Context.csproj [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  <PropertyGroup>
+    <UseECMAKey Condition="'$(UseECMAKey)' == ''">true</UseECMAKey>
+    <AssemblyVersion>4.0.1.0</AssemblyVersion>
+    <OutputType>Library</OutputType>
+    <PackageTargetFramework>dotnet5.2</PackageTargetFramework>
+    <NuGetTargetMoniker>.NETPlatform,Version=v5.2</NuGetTargetMoniker>
+    <ProjectGuid>{404DB891-B5AF-41E6-B89D-29E3F4573C4F}</ProjectGuid>
+  </PropertyGroup>
+  <!-- Default configurations to help VS understand the options -->
+  <PropertyGroup Condition="'$(Configuration)|$(Platform)' == 'netcore50_Debug|AnyCPU'" />
```
Comment (weshaggard) [inline src/System.Reflection.Context/src/System.Reflection.Context.csproj]: netcore50_Debug => Windows_netcore50_Debug

**Comment by weshaggard** [comment]
Just the one small tweak to the configuration list otherwise LGTM.

**Comment by mellinoe** [comment]
@weshaggard Made the configuration changes. Tests are re-running now.

**Comment by mellinoe** [comment]
@dotnet-bot Test Innerloop CentOS7.1 Debug Build and Test

**Comment by mellinoe** [comment]
@dotnet-bot Test Innerloop CentOS7.1 Release Build and Test

**Pull request merged**